var/spack/repos/builtin/packages/hdf/package.py

```diff
@@ -15,45 +15,121 @@ class Hdf(AutotoolsPackage):
     list_url = "https://support.hdfgroup.org/ftp/HDF/releases/"
     list_depth = 2
 
+    version('4.2.15', sha256='dbeeef525af7c2d01539906c28953f0fdab7dba603d1bc1ec4a5af60d002c459')
     version('4.2.14', sha256='2d383e87c8a0ca6a5352adbd1d5546e6cc43dc21ff7d90f93efa644d85c0b14a')
     version('4.2.13', sha256='be9813c1dc3712c2df977d4960e1f13f20f447dfa8c3ce53331d610c1f470483')
     version('4.2.12', sha256='dd419c55e85d1a0e13f3ea5ed35d00710033ccb16c85df088eb7925d486e040c')
     version('4.2.11', sha256='c3f7753b2fb9b27d09eced4d2164605f111f270c9a60b37a578f7de02de86d24')
 
     variant('szip', default=False, description="Enable szip support")
-    variant('libtirpc', default=False, description="Use xdr library from libtirpc package; if false, will use system or hdf internal")
+    variant('xdr', default='system', values=('system', 'embedded', 'external'),
+            description="Specify XDR backend")
+    variant('netcdf', default=False,
+            description='Build NetCDF API (version 2.3.2)')
+    variant('fortran', default=False,
+            description='Enable Fortran interface')
+    variant('java', default=False,
+            description='Enable Java JNI interface')
+    variant('shared', default=False, description='Enable shared library')
+    variant('pic', default=True,
+            description='Produce position-independent code')
 
-    depends_on('jpeg@6b:')
-    depends_on('szip', when='+szip')
-    depends_on('libtirpc', when='+libtirpc')
     depends_on('zlib@1.1.4:')
+    depends_on('jpeg')
+    depends_on('szip', when='+szip')
+    depends_on('libtirpc', when='xdr=external')
 
     depends_on('bison', type='build')
     depends_on('flex',  type='build')
+    depends_on('java@7:', when='+java', type=('build', 'run'))
 
-    def configure_args(self):
-        spec = self.spec
-
-        config_args = [
-            'CFLAGS={0}'.format(self.compiler.cc_pic_flag),
-            '--with-jpeg={0}'.format(spec['jpeg'].prefix),
-            '--with-zlib={0}'.format(spec['zlib'].prefix),
-            '--disable-netcdf',  # must be disabled to build NetCDF with HDF4
-            '--enable-fortran',
-            '--disable-shared',  # fortran and shared libs are not compatible
-            '--enable-static',
-            '--enable-production'
-        ]
-
-        # Szip support
-        if '+szip' in spec:
-            config_args.append('--with-szlib={0}'.format(spec['szip'].prefix))
+    # https://forum.hdfgroup.org/t/cant-build-hdf-4-2-14-with-jdk-11-and-enable-java/5702
+    patch('disable_javadoc.patch', when='@:4.2.14+java')
+
+    conflicts('^libjpeg@:6a')
+
+    conflicts('+shared', when='+fortran')
+
+    conflicts('+java', when='~shared')
+    conflicts('+java', when='@:4.2.11')
+    conflicts('+java', when='@4.2.12:4.2.13~netcdf')
```

Comment:
Maybe add messages to some of these conflicts explaining why they are necessary?

Reply:
done